docs/ADR-58-third-party-assets-integration-v2-with-merkle-tree.md

```diff
@@ -0,0 +1,163 @@
+
+```solidity
+struct ThirdParty {
+        bool isApproved;
+        bytes32 root; // new
+        uint256 maxItems;
+        uint256 consumedSlots; // new
+        uint256 registered;
+        string metadata;
+        string resolver;
+        string[] itemIds;
+        mapping(bytes32 => uint256) receipts; // new
+        mapping(address => bool) managers;
+        mapping(string => Item) items;
+        mapping(string => bool) rules; // new
+    }
+```
+
+- _`isApproved`_: whether a third party is approved or not.
+- _`root`_: current merkle tree root. This root is going to be used by the Decentraland catalyst to check if an item deployment is valid or not.
+- _`maxItems`_: represents the maximum number of items that a third party can have. We call them _item slots_. Item slots can be bought by everyone at any time, on multiple occasions, by using MANA. So, it is not necessary to be a third party manager to buy item slots. Item slots are going to be bought by unit. The price is set in USD dollar defined in the _`itemSlotPrice`_ variable.
+- _`consumedSlots`_: the number of items consumed by a third party.
+- _`registered`_: simple boolean that helps to check whether a third party has been added or not.
+- _`metadata`_: string with the following shape: `type:version:name:description`. i.e: `tp:1:third party 1:the third party 1 description`. We may include collections metadata by changing the metadata version.
+- _`resolver`_: string with the third party API resolver. This API will be used for services to get which Decentraland asset should be mapped to which NFT token. _We call Decentraland asset to every asset that is submitted to the Decentraland catalyst_. i.e: `https://api.thirdparty1.com/v1/get-owned-nfts/:owner`
+- _`itemIds`_: in order to allow looping through the items added without the need of indexing historic events, we need to keep track of their ids. **IMPORTANT**: Items reviewed by using a merkle tree root, are not going to be submitted to the blockchain.
+- _`receipts`_: mapping of item consumptions messages hash. We will use this in order to prevent double-spending them.
+- _`managers`_: third party managers.
+- _`items`_: third party's items. **IMPORTANT**: Items reviewed by using a merkle tree root, are not going to be submitted to the blockchain.
+- _`rules`_: third party's rules. **IMPORTANT**: Rules are not going to be used for the time being. The rejection of third parties and their items are still a pending topic.
+
+A third party record can't be removed but approved/rejected by a committee member.
+
+Third parties can be looped off-chain without the need of indexing historic events.
+
+### Items
+
+Items are encourage to not be submitted to the blockchain in order to reduce the number of transactions. We will keep the old way of submitting them but we may remove it in a near future. The only way available to loop throught the third parties' items is by using the Decentraland catalysts:
+
+// TODO: add endpoint example
```

Comment:
@agusaldasoro can you point me here what is going to be the lambda to get the items from a third party/collection

Reply:
Do you mean the filter per urn prefix? Here: https://github.com/decentraland/catalyst-api-specs/pull/21/files